neat_stat.c

```diff
@@ -0,0 +1,36 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "neat_internal.h"
+#include "neat_core.h"
+#include "neat_stat.h"
+
+
+/* Traverse the relevant subsystems of NEAT and gather the stats
+   then format the stats as a json string to return */	
+void neat_stats_build_json(neat_flow *flow, char **json_stats)
+{
+	neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+	json_t *root = json_object();
```

Comment:
I'll switch the place of the statements so the debug printout comes after
declarations.

On 15 June 2016 at 11:18, Kristian A. Hiorth <notifications@github.com>
wrote:

> In neat_stat.c
> <https://github.com/NEAT-project/neat/pull/90#discussion_r67126789>:
>
> > +#include <stdarg.h>
> > +#include <stdlib.h>
> > +#include <string.h>
> > +
> > +#include "neat_internal.h"
> > +#include "neat_core.h"
> > +#include "neat_stat.h"
> > +
> > +
> > +/* Traverse the relevant subsystems of NEAT and gather the stats
> > +   then format the stats as a json string to return */	
> > +void neat_stats_build_json(neat_flow *flow, char **json_stats)
> > +{
> > +	neat_log(NEAT_LOG_DEBUG, "%s", __func__);
> > +
> > +	json_t *root = json_object();
>
> I think we reached concensus that we should avoid declaring variables
> within blocks, maybe move declaration to top?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NEAT-project/neat/pull/90/files/24b0d23d9732e1813e296d2d221beb851bec481a#r67126789>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACejY-ZfJxAdtJPe0QT6EZV91vftM2Pyks5qL8N5gaJpZM4Izozz>
> .
>